tools/autobuild/build-stm32-extra.sh

```diff
@@ -7,12 +7,15 @@ function do_build() {
     board=$2
     shift
     shift
-    echo "building $descr $board"
-    build_dir=/tmp/stm-build-$board
-    $MICROPY_AUTOBUILD_MAKE $@ BOARD=$board BUILD=$build_dir || exit 1
-    mv $build_dir/firmware.dfu $dest_dir/$descr$fw_tag.dfu
-    mv $build_dir/firmware.hex $dest_dir/$descr$fw_tag.hex
-    rm -rf $build_dir
+    for variant in `$MICROPY_AUTOBUILD_MAKE BOARD=$board query-variants | grep VARIANTS: | cut -d' ' -f2-`; do
+        target=$descr-$variant
+        echo "building $target $board"
+        build_dir=/tmp/stm-build-$board
+        echo $MICROPY_AUTOBUILD_MAKE $@ BOARD=$board BOARD_VARIANT=$variant BUILD=$build_dir || exit 1
+        echo mv $build_dir/firmware.dfu $dest_dir/$descr$fw_tag.dfu
```

Comment:
the echo's need to be deleted, and `$descr` should be `$target`

Reply:
Thanks. Rebased and updated.